pkg/services/promtypemigration/prom_mig.go

```diff
@@ -0,0 +1,89 @@
+package promtypemigration
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/repo"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type PromMigrationHandler interface {
+	Migrate(context.Context, *promMigrationService) error
+	getPrometheusDataSources(ctx context.Context, dataSourcesService datasources.DataSourceService) ([]*datasources.DataSource, error)
+}
+
+type dataSourceService interface {
+	DecryptedValues(ctx context.Context, ds *datasources.DataSource) (map[string]string, error)
+	UpdateDataSource(ctx context.Context, cmd *datasources.UpdateDataSourceCommand) (*datasources.DataSource, error)
+	GetDataSourcesByType(ctx context.Context, query *datasources.GetDataSourcesByTypeQuery) ([]*datasources.DataSource, error)
+}
```

Comment:
can you use the [exported interface](https://github.com/grafana/grafana/blob/andrew/prom-type-migration-service/pkg/services/datasources/datasources.go#L14)?

Reply:
I can but then I have to mock a bunch of extra stuff in the tests right?

Reply:
it's not mandatory, you can define a `struct` that doesn't define all the methods but if you add the interface on it, it won't complain:

```
type MockDSService struct {
	datasources.DataSourceService
}
```